File: AngularJS/src/org/angularjs/html/Angular2HTMLParserDefinition.java
package org.angularjs.html;

import com.intellij.ide.highlighter.HtmlFileType;
import com.intellij.lang.ASTNode;
import com.intellij.lang.html.HTMLParserDefinition;
import com.intellij.lexer.Lexer;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.openapi.project.Project;
import com.intellij.psi.FileViewProvider;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFile;
import com.intellij.psi.impl.source.html.HtmlFileImpl;
import com.intellij.psi.tree.IFileElementType;
import com.intellij.psi.tree.IStubFileElementType;
import org.jetbrains.annotations.NotNull;

/**
 * @author Dennis.Ushakov
 */
public class Angular2HTMLParserDefinition extends HTMLParserDefinition {
  static IFileElementType HTML_FILE = new IStubFileElementType(Angular2HTMLLanguage.INSTANCE);

  @NotNull
  @Override
  public Lexer createLexer(Project project) {
    return new Angular2HTMLLexer();
  }

  @NotNull
  @Override
  public PsiElement createElement(ASTNode node) {
    return super.createElement(node);
  }

  @Override
  public IFileElementType getFileNodeType() {
    return HTML_FILE;
  }

  @Override
  public PsiFile createFile(FileViewProvider viewProvider) {
    return new HtmlFileImpl(viewProvider, HTML_FILE) {
      @NotNull
      @Override
      public FileType getFileType() {
        return HtmlFileType.INSTANCE;
      }
    };
  }
}

File: Angular/src/org/angular2/lang/html/parser/Angular2HtmlParserDefinition.kt
// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package org.angular2.lang.html.parser

import com.intellij.psi.tree.IFileElementType
import org.angular2.lang.html.Angular2HtmlFileElementType
import org.angular2.lang.html.Angular2TemplateSyntax

open class Angular2HtmlParserDefinition : Angular2HtmlParserDefinitionBase() {

  override val syntax: Angular2TemplateSyntax
    get() = Angular2TemplateSyntax.V_2

  override fun getFileNodeType(): IFileElementType {
    return Angular2HtmlFileElementType.INSTANCE
  }
}